src/AbstractActionHandler.ts

```diff
@@ -166,12 +169,21 @@ export abstract class AbstractActionHandler {
     versionedActions: VersionedAction[],
     block: Block,
     context: any,
+    lastIrreversibleBlockNumber: number,
   ) {
+    this.runDeferredEffects(lastIrreversibleBlockNumber)
     for (const { action, handlerVersionName } of versionedActions) {
       for (const effect of this.handlerVersionMap[handlerVersionName].effects) {
         if (this.matchActionType(action.type, effect.actionType)) {
           const { payload } = action
-          effect.run(payload, block, context)
+          if (!effect.deferUntilIrreversible || block.blockInfo.blockNumber <= lastIrreversibleBlockNumber) {
```

Comment:
This code might be easily refactored to like a `updateDeferredEffects` method or something like that

Reply:
Yeah, had the same thought.